packages/block-editor/src/components/fit-text-size-warning/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Component that checks if the fit text computed font size is below a minimum
+ * threshold and displays a warning notice.
+ *
+ * @return {Element|null} Warning notice element or null.
+ */
+export default function FitTextSizeWarning() {
+	const message = __(
+		'The text may be too small to read. Consider using a larger container or less text.'
+	);
+
+	// Note: The `Notice` component can speak messages via its `spokenMessage`
+	// prop, but similar to the contrast checker, we use granular control over
+	// when the announcements are made.
+	speak( message );
```

Comment:
I haven't tested this but, wouldn't this give information to screen readers _in every render_? I wonder if it'd be best to use this only when mounting the component. 🤔